### DISCUSSION TITLE
[@types/react] What is the process for upgrading several types across react versions?

### DISCUSSION THREAD

**Discussion by lifeiscontent** [DefinitelyTyped/DefinitelyTyped#61494] [Issues with a @types package] [answered]
I asked this question here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61473 but I'm guessing since the issue was closed, no one is going to respond at this point.

There are several HTML elements that are using the wrong interfaces and wanted to update that, as well as make props/state ReadOnly as I assume it should be.

is there a process in place for upgrading types across versions? or what has been the method for updating all versions of React?

@eps1lon do you have some suggestions here?

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
> is there a process in place for upgrading types across versions?

There's no specifc process. Just update the types for older versions. Older versions are located in the `v*` folders e.g. `v16` is for `react@16.x`.

**Reply by lifeiscontent** [reply]
@eps1lon got it, thank you for your response!
